test/common_distributed.py

```diff
@@ -146,8 +150,21 @@ def _join_processes(self, fn):
         for p in self.processes:
             p.join(timeout)
         elapsed_time = time.time() - start_time
-        if fn in self.skip_return_code_checks:
+        if fn not in self.skip_return_code_checks:
             self._check_return_codes(elapsed_time)
+        else:
+            # Check no assertion errors.
+            self._check_no_assertion_error(elapsed_time)
+
+    def _check_no_assertion_error(self, elapsed_time):
+        """
+            Checks that we didn't have any assertion errors thrown in the child
+            processes.
+        """
+        for i, p in enumerate(self.processes):
+            if p.exitcode is None:
+                raise RuntimeError('Process {} terminated or timed out after {} seconds'.format(i, elapsed_time))
```

Comment:
This condition is only true if the call to `join` timed out. It won't have terminated here.

Reply:
Probably an oversight. Commenting here because the lines changed :)